### PR TITLE
New compiler

### PR DESCRIPTION
**Pull request by zhfnjust**

### CODE REVIEW

File: contracts/tsp.scrypt [hunk 1]
```diff
@@ -1,32 +1,36 @@
 // travelling salesman problem
 contract TSP {
+
     int threshold;
+
     // 10 * 10 adjacent matrix repressentation
     int[100] graph;
 
-    public function verify(int[10] path) {
+    static const int N = 10;
```
Comment (xhliu) [inline contracts/tsp.scrypt:9]:
```
  // N * N adjacent matrix repressentation
    int[NSQUARE] graph;

 static const int N = 10;
 static const int NSQUARE = 100;
```

File: contracts/tictactoe.scrypt [hunk 2]
```diff
@@ -79,7 +79,7 @@ contract TicTacToe {
         bool full = true;
         int i = 0;
 
-        loop (9) {
+        loop (BOARDLEN) {
             if (full) {
                 if (Util.getElemAt(board, i) == TicTacToe.EMPTY) {
```
Comment (xhliu) [inline contracts/tictactoe.scrypt:84]: remove all `TicTacToe.`

File: contracts/sudoku.scrypt [hunk 3]
```diff
@@ -3,7 +3,7 @@ contract Sudoku {
```
Comment (xhliu) [inline contracts/sudoku.scrypt:3]: Let's not check in sudoku till u fix it.

File: contracts/serializer.scrypt [hunk 4]
```diff
@@ -48,7 +48,7 @@ contract Reader {
```
Comment (xhliu) [inline contracts/serializer.scrypt:48]: All
this.pos = this.pos + 1;
-> 
this.pos++;


this.pos = this.pos + len;
->
this.pos += len;

File: contracts/tictactoe.scrypt [hunk 5]
```diff
@@ -79,7 +79,7 @@ contract TicTacToe {
         bool full = true;
         int i = 0;
 
-        loop (9) {
+        loop (BOARDLEN) {
```
Comment (xhliu) [inline contracts/tictactoe.scrypt:82]: Line 88
i = i + 1;
->
i++;

**Review by xhliu — CHANGES_REQUESTED** [review]
👍 

Let's update all syntax to latest compiler.

File: contracts/demo.scrypt [hunk 6]
```diff
@@ -1,21 +1,22 @@
 contract Demo {
-  int x;
-  int y;
+    int x;
```
[the remark this answers is not in the document]
Comment (zhfnjust) [reply contracts/demo.scrypt:2]: format space change from 2 to 4

**Pull request merged**